### ISSUE TITLE
Add request decompression middleware

### ISSUE THREAD

**Issue by matkob** [http4s/http4s#7072]
Hello everyone!

I'm using `GZip` middleware in one of the projects and recently realized that it only supports response compression. My idea is to either add an extra capability to decompress the requests as well, or create another middleware for that.

What do you think? I've already written the code and it can be merged with current `GZip` implementation fairly easily.

**Comment by armanbilge** [comment]
Yup, contribution would be welcome here! :) We might have some similar issues / stale PRs lying around, can't remember, but no matter, happy to see this land. Thanks!

**Comment by rossabaker** [comment]
Related and stale discussion on https://github.com/http4s/http4s/issues/4981.

**Comment by matkob** [comment]
Thank you for the link, I'll catch up and prepare the code. 
I saw bunch of open discussions there and am wondering what would be the preferred way for future ones, shall I continue here or move to the old one?

**Comment by rossabaker** [comment]
I'd say if it's related to that ticket, we can keep discussing there, where there's more research, even if it's old.  Thanks!
